### PR TITLE
Add WithQueryParameters to policy package

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
Enables custom query params for API calls.

Fixes https://github.com/Azure/azure-sdk-for-go/issues/21704

### CODE REVIEW

**Comment by jhendrixMSFT** [comment]
@haitch FYI

**Comment by JeffreyRichter** [comment]
The code looks fine but why are we adding this? Is there some customer scenario that requires this?

**Comment by jhendrixMSFT** [comment]
AKS folks asked for this.

**Comment by JeffreyRichter** [comment]
OK but I'd still like to know more. If the query params are in swagger/TypeSpec, then we would have them in the SDK. So, do they have undocumented query parameters that they need this for? And, if so, this is not ideal.

**Comment by jhendrixMSFT** [comment]
Good question.  @serbrech can you elaborate on your scenario?

File: sdk/azcore/runtime/policy_query_params.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+		var err error
+		allQP, err = url.ParseQuery(req.Raw().URL.RawQuery)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	newQP := rawQP.(url.Values)
+	if len(allQP) > 0 {
+		// merge the existing query params with the ones in the context.
```
Comment (haitch) [inline sdk/azcore/runtime/policy_query_params.go:37]: this merge might have a issue, it's always replace behavior.

http allows queryparameter to be repeated, like ?name=bob&name=alice. 

let's say I have name=bob in my original request, I want to append name=alice, this policy won't work, it always replace.

**Review by haitch — COMMENTED** [review]
overall, it will work for us. I am not going to block it, but maybe there could be case where this is a issue.

**Comment by jhendrixMSFT** [comment]
Closing for now until we work out a few more details.

**Pull request closed without merging**